### ISSUE TITLE
Calling for volunteers for adding operators!

### ISSUE THREAD

**Issue by HAOCHENYE** [open-mmlab/mmcv#2409]
# Task description

As MMCV supports more and more downstream repositories, more and more operators are added to MMCV. However, not all operators are adapted to all hardware, for example, we only provide `softnms` with CPU implementation but not GPU implementation, and focal loss only with GPU implementation but not CPU implementation.

In order to make it easier for everyone to run MMCV on different devices, we call on everyone to participate in this operator implementation activity, from which you will learn:

1. The whole process of PR from creation to merging
2. Be familiar with and understand the process of adding arithmetic to MMCV
3. Know how to extend PyTorch at the C++/CUDA level

If you are interested, you can claim a operators by replying to this issue for the following format:

```
The operator name you claim
1. The Repo links your fork
2. The PR links you open(Update this after posting the PR)
```

The full PR process can be referred to the [guide](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md).

| Operator name | device | Level of Difficulty | Assigned to | Status | 
| :-: | :-: | :-: | :-: | :-: |
| [softnms](#softnms) | CUDA | ⭐️⭐️⭐️ |  |  |
| [nms_match](#nms_match) | CUDA | ⭐️⭐️⭐️ |  |  |
| [sigmoid_focal_loss](#sigmoid_focal_loss) | CPU | ⭐️⭐️⭐️ |  |  |
| [softmax_focal_loss](#softmax_focal_loss) | CPU | ⭐️⭐️⭐️ |  |  |
| [assign_score_withk](#assign_score_withk) | CPU | ⭐️⭐️⭐️ |  |  |
| [bbox_overlaps](#bbox_overlaps) | CPU | ⭐️⭐️⭐️ |  |  |
| [correlation](#correlation) | CPU | ⭐️⭐️⭐️⭐️ |  |  |
| [ms_deform_attn](#ms_deform_attn) | CPU | ⭐️⭐️⭐️⭐️⭐️ |  |  |

# Template PR

[2365](#2365) provide the template for [softnms](#softnms), [sigmoid_focal_loss](#sigmoid_focal_loss) and [softmax_focal_loss](#softmax_focal_loss). It only finish the registering, dispatching, and upgrading unit test process, you need to implement the function which only implements `Please implement xxx`

# Operators

## softnms

- CPU Reference Implementation:

    https://github.com/open-mmlab/mmcv/blob/652b1bf207616608179398abdcf17276b4a0ab27/mmcv/ops/csrc/pytorch/cpu/nms.cpp#L59

- Unit test to be updated:

    https://github.com/open-mmlab/mmcv/blob/652b1bf207616608179398abdcf17276b4a0ab27/tests/test_ops/test_nms.py#L41

- Technical Tags
   C++; CUDA; Python; Detection

## nms_match

- CPU Reference Implementation:

    https://github.com/open-mmlab/mmcv/blob/652b1bf207616608179398abdcf17276b4a0ab27/mmcv/ops/csrc/pytorch/cpu/nms.cpp#L168

- Unit test to be updated:

    https://github.com/open-mmlab/mmcv/blob/652b1bf207616608179398abdcf17276b4a0ab27/tests/test_ops/test_nms.py#L110

- Technical Tags
   C++; CUDA; Python; Detection

## sigmoid_focal_loss

- CUDA Reference Implementation:

    https://github.com/open-mmlab/mmcv/blob/652b1bf207616608179398abdcf17276b4a0ab27/mmcv/ops/csrc/common/cuda/sigmoid_focal_loss_cuda_kernel.cuh#L12

- Unit test to be updated:

    https://github.com/open-mmlab/mmcv/blob/652b1bf207616608179398abdcf17276b4a0ab27/tests/test_ops/test_focal_loss.py#L63

- Technical Tags
   C++; CUDA; Python

## softmax_focal_loss

- CUDA Reference Implementation:

    https://github.com/open-mmlab/mmcv/blob/652b1bf207616608179398abdcf17276b4a0ab27/mmcv/ops/csrc/common/cuda/softmax_focal_loss_cuda_kernel.cuh#L12

- Unit test to be updated:

    https://github.com/open-mmlab/mmcv/blob/652b1bf207616608179398abdcf17276b4a0ab27/tests/test_ops/test_focal_loss.py#L42

- Technical Tags
   C++; CUDA; Python

## assign_score_withk

- CUDA Reference Implementation:

    https://github.com/open-mmlab/mmcv/blob/652b1bf207616608179398abdcf17276b4a0ab27/mmcv/ops/csrc/common/cuda/assign_score_withk_cuda_kernel.cuh#L20

- Unit test to be updated:

    https://github.com/open-mmlab/mmcv/blob/652b1bf207616608179398abdcf17276b4a0ab27/tests/test_ops/test_assign_score_withk.py#L10

- Technical Tags
   C++; CUDA; Python

## bbox_overlaps

- CUDA Reference Implementation:

    https://github.com/open-mmlab/mmcv/blob/652b1bf207616608179398abdcf17276b4a0ab27/mmcv/ops/csrc/common/cuda/bbox_overlaps_cuda_kernel.cuh#L33

- Unit test to be updated:

    https://github.com/open-mmlab/mmcv/blob/652b1bf207616608179398abdcf17276b4a0ab27/tests/test_ops/test_bbox.py#L9

- Technical Tags
   C++; CUDA; Python; Detection

## correlation

- CUDA Reference Implementation:

    https://github.com/open-mmlab/mmcv/blob/652b1bf207616608179398abdcf17276b4a0ab27/mmcv/ops/csrc/common/cuda/correlation_cuda.cuh#L36

- Unit test to be updated:

    https://github.com/open-mmlab/mmcv/blob/652b1bf207616608179398abdcf17276b4a0ab27/tests/test_ops/test_correlation.py#L19

- Technical Tags
   C++; CUDA; Python; Detection

## ms_deform_attn

- CUDA Reference Implementation:

    https://github.com/open-mmlab/mmcv/blob/652b1bf207616608179398abdcf17276b4a0ab27/mmcv/ops/csrc/common/cuda/ms_deform_attn_cuda_kernel.cuh#L18

- Unit test to be updated:

    https://github.com/open-mmlab/mmcv/blob/652b1bf207616608179398abdcf17276b4a0ab27/tests/test_ops/test_ms_deformable_attn.py#L107

- Technical Tags
   C++; CUDA; Python; Detection

**Comment by ahmedmustahid** [comment]
Is it only open for residents in China? The "discord" server QR code above directs me to We Chat. Also it requires a Chinese phone number to sign up using the [application form link](https://openmmlab.com/activity/codecamp/apply) provided above.

**Comment by zhouzaida** [comment]
Hi @ahmedmustahid , thank you very much for your interest in our activities, you can join our discussion group by clicking https://discord.gg/KuWMWVbCcD.
